### PR TITLE
The new README

### PR DESCRIPTION
**Pull request by vkazanov**
As discussed in #70 I tried my best to introduce a better structure to the file.

A minor additional tweak: I've added a C-c C-s binding for org-journal-search in the org-journal-mode.

### CODE REVIEW

File: README.org [hunk 1]
```diff
@@ -1,88 +1,23 @@
...
 
-Functions to maintain a simple personal diary / journal in Emacs.
-Feel free to use, modify and improve the code!
-— mtvoid, bastibe
+  Adapted from http://www.emacswiki.org/PersonalDiary
 
-*NEWS*
+  Functions to maintain a simple personal diary / journal using in Emacs.
+  Feel free to use, modify and improve the code!
+  — mtvoid, bastibe
```
Comment (bastibe) [inline README.org]: An explicit line break would be nice, here.

File: README.org [hunk 2]
```diff
@@ -1,88 +1,23 @@
...
 
-This file is also available from [[http://marmalade-repo.org/][marmalade]], [[http://melpa.milkbox.net/][melpa]] and [[http://melpa-stable.milkbox.net/][melpa-stable]]
-under the name org-journal. You may also want to specify the directory
-where your journal files will be saved. You can do this by setting the
-variable =org-journal-dir= (remember to add a trailing slash).
-=org-journal-dir= is also a customizable variable. The default value
-for =org-journal-dir= is =~/Documents/journal/=.
+   =org-journal= maintains a set of files where each file represents a day. Convenient bindings allow
+   to create journal records in the current daily file and search within all records or specified
+   time intervals. All records can be browsed and searched from the Emacs Calendar for convenience.
```
Comment (bastibe) [inline README.org]: "...allow the creation of journal records..."

File: README.org [hunk 3]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+    C-x package-install org-journal
+#+END_EXAMPLE
+
+   Then add =(require 'org-journal)= to your =.emacs=.
+
+** Quickstart
+
+   Doing =C-x org-journal-new-entry= (or =C-c C-j=) will immediately create a journal directory in the
+   default path (customized using the =org-journal-dir= variable), open a file in =org-journal-mode= and
+   insert a journal entry template.
```
Comment (bastibe) [inline README.org]: "...open or create a file...", "...and insert a template for a new journal entry."

File: README.org [hunk 4]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+
+   Same command with a prefix argument (=C-u C-c C-j=) will do everything mentioned while skipping
```
Comment (bastibe) [inline README.org]: "The same..."

File: README.org [hunk 5]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+   entry creation, which is useful for taking a look at the current journal file.
+
+** Basic Usage
+
+   Bindings available in =org-journal-mode=:
+
+   - =C-c C-f= - find next entry
```
Comment (bastibe) [inline README.org]: "go to the next journal file". I believe you used "entry" for level-2 headlines before.

File: README.org [hunk 6]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+
+   - =C-c C-b= - find previous entry
```
Comment (bastibe) [inline README.org]: "go to the previous journal file"

File: README.org [hunk 7]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+
+   - =C-c C-j= - insert a new entry
```
Comment (bastibe) [inline README.org]: "...into the current journal file (creates the file if not present)"

File: README.org [hunk 8]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+
+   - =C-c C-s= - search the journal for a string
+
+   All journal entries are registered in the Emacs Calendar. To see available journal entries do =C-x
+   calendar=. Bindings available in the =calendar-mode=:
+
+   - =j= - view an entry in a new buffer
+
+   - =C-j= - view an entry but not switch to it
```
Comment (bastibe) [inline README.org]: "...but do not..."

File: README.org [hunk 9]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+
+   - =i j= - add a new entry
```
Comment (bastibe) [inline README.org]: "...into the day's file (creates the file if not present)"

File: README.org [hunk 10]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
-  Initial working prototype. More or less simply an extension of personal-diary with org-mode files.
+   - =f y= - search in all entries of the current year
 
-- 2013-06-12 V1.3.4 Bastian Bechtold ::
-  Several bug fixes and documentation improvements.
+   - =f f= - search in all entries of all time
 
-- 2013-08-23 V1.4 Bastian Bechtold ::
-  Now supports more customization
+   - =[= - go to previous entry
```
Comment (bastibe) [inline README.org]: "go to previous day with journal entries"

File: README.org [hunk 11]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
 
-- 2013-09-17 V1.4.3 Bastian Bechtold ::
-  Several bugfixes and autoloading (which kind of mostly works)
+   - =]= - go to next entry
```
Comment (bastibe) [inline README.org]: "go to next day with journal entries"

File: README.org [hunk 12]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
 
-- 2014-03-10 V1.4.8 Bastian Bechtold ::
-  Some bug fixes on opening files (thank you, Ram Raghunathan!)
+    Customization options related to journal directory and files:
 
-- 2014-06-17 V1.5 Bastian Bechtold ::
-  Some more bug fixes for autoloading
-  Simplified setup by auto generating the file pattern from the file format.
+    - =org-journal-dir= - the journal path. Tweaking this variable will also update =auto-mode-alist= to
+      make sure journal files are handled using the correct mode.
```
Comment (bastibe) [inline README.org]: "...are opened in org-journal-mode"

File: README.org [hunk 13]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+    =org-read-date= format such as "-1y" or "-1m") and the string to search for. Given a prefix
+    argument (=C-u org-journal-search=) it will go through the whole journal.
+
+    Search is also available through the Emacs Calendar as described in "Basic Usage".
+
+*** Carry Over
+
+    By default =org-journal= will try to /carry over/ previous day TODO-marked items whenever a new
+    journal file is created. The older journal entry will be /moved/ (i.e., deleted and reinserted) to
+    the current day file.
```
Comment (bastibe) [inline README.org]: "the current day's file."

File: README.org [hunk 14]
```diff
@@ -114,63 +49,133 @@ A typical journal entry for a day would look like this:
...
+
+    This feature is controlled through the =org-journal-carryover-items= variable. Set this to =nil= to
+    disable this feature, or to any [[http://orgmode.org/manual/Matching-tags-and-properties.html][agenda tags view match string]] for a set of matching tags,
+    properties, and todo states. By default, this is ~TODO=”TODO”~, which will match TODO items.
+
+*** Encryption
+
+    The journal can be encrypted using =org-crypt= when ~org-journal-enable-encryption~ is non-nil.
+
+** FAQ
```
Comment (bastibe) [inline README.org]: Very good idea to include this section!

File: org-journal.el [hunk 15]
```diff
@@ -211,6 +211,7 @@ Otherwise, date ascending."
 (define-key org-journal-mode-map (kbd "C-c C-f") 'org-journal-open-next-entry)
 (define-key org-journal-mode-map (kbd "C-c C-b") 'org-journal-open-previous-entry)
 (define-key org-journal-mode-map (kbd "C-c C-j") 'org-journal-new-entry)
+(define-key org-journal-mode-map (kbd "C-c C-s") 'org-journal-search)
```
Comment (bastibe) [inline org-journal.el]: great idea!

**Comment by vkazanov** [comment]
Great, I'll fix those ASAP

**Comment by vkazanov** [comment]
Done :-)

**Comment by bastibe** [comment]
OK, very cool! As I said, I'll have a native speaker read over this and report back. This might take a day or two.

**Comment by vkazanov** [comment]
Cool! The more polish it gets the better. 

Just a side note: I wonder if it'll be possible to see how the new README influences people checking out the project. :-)

**Comment by bastibe** [comment]
There are download stats on melpa that put org-journal in the top 17th percentile. This only counts the overall number of downloads though, and does not track changes over time. However, if the percentile would slide up, we could count that as a win.

**Comment by vkazanov** [comment]
Hm. That should do :-)

**Comment by vkazanov** [comment]
I've added a couple more thingies into the README.

**Comment by bastibe** [comment]
I just got the review from my native speaker friend, but I could not figure out how to apply it to the pull request without copying her changes into a review, and then have you copy those changes into the file. Too much copying.

I'll apply her changes separately.

Anyway, thank you very much for this! I believe that the README was indeed substantially improved!

**Comment by vkazanov** [comment]
Nice, thanks!

**Pull request merged**